mqtt/mqtt-edgehub/src/auth/authorization/edgehub.rs

```diff
@@ -266,6 +275,11 @@ where
                 self.inner.update(update)?;
             }
         };
+
+        if let Some(mut broker_ready) = self.broker_ready.take() {
```

Comment:
I think you need to move this into match - Ok case. It is possible that PolicyUpdate comes first, and would cause false-positive ready signal.

Reply:
oops, that's right. good catch!